### ISSUE TITLE
Show last changes view for intents

### ISSUE THREAD

**Issue by turnerrainer** [buerokratt/Training-Module#49]
AS A Chatbot Trainer
I WANT TO know when the last changes were made to a specific intent
SO I WOULD know when it was last changed

#### DSL

- [ ] Ruuter GET endpoint `/intents/history/last-modified/{intentID)`
> If Ruuter does not accept {intentID} as a parameter when doing this issue, use POST service instead
- [ ] Intents with metadata and time last changed in Postgres table
- [ ] Resql endpoint to fetch appropriate data

#### GUI

- [ ] According to [prototype](http://proto.we.ee/burokratt/treening/treening/teemad)

![image](https://user-images.githubusercontent.com/90505798/211764420-b2ef9f1b-b543-4fbb-8d37-fe5ebfe7ad1e.png)

**Comment by RaulAltmae** [comment]
POST endpoint `/intents/history/last-modified/{intentID)`
with input
```json
{
    "intent": "common_teenus_ilm"
}
```

**Comment by turnerrainer** [comment]
@RaulAltmae please link this issue with an appropriate pull request.

**Comment by RaulAltmae** [comment]
Pull request https://github.com/buerokratt/Training-Module/pull/147

**Comment by turnerrainer** [comment]
Moving this one as `Blocked` at the moment as there are no developments making entries to `intent_last_changed`.

**Comment by turnerrainer** [comment]
@varmoh need to make sure that Liquibase changes have been applied.

**Comment by varmoh** [comment]
Created a script to deploy Postgresql database and liquibase seeding. Can be used to deploy a standalone database or, with small change, existing database seeding.

**Comment by turnerrainer** [comment]
> Created a script to deploy Postgresql database and liquibase seeding. Can be used to deploy a standalone database or, with small change, existing database seeding. For testing, should I create a standalone database for Training-module or seed the required intents field into existing database ?

@varmoh let's have a separate, standalone database for Training Module with a Postgres deploy and Liquibase seeding pipelines.

**Comment by turnerrainer** [comment]
> > Created a script to deploy Postgresql database and liquibase seeding. Can be used to deploy a standalone database or, with small change, existing database seeding. For testing, should I create a standalone database for Training-module or seed the required intents field into existing database ?
> 
> @varmoh let's have a separate, standalone database for Training Module with a Postgres deploy and Liquibase seeding pipelines.

@varmoh the above should be a separate issue under NoOps. If the functionality of current issue is OK, mark it as `Done` please.

**Comment by turnerrainer** [comment]
Blocked by #289

**Comment by KasparMinn** [comment]
PR: https://github.com/buerokratt/Training-Module/pull/291 

This pull request finalizes and fixes any changes needed to properly parse and display the 'last changed' date of any selected intent in the GUI. This data is rather minute, a response with a single field. It gets automatically queried whenever a new tab is selected. If the value can be properly parsed into a date and becomes 'truthy', it is shown. If for some reason the date is unavailable, the GUI will keep working and display 'unknown'.

In addition to that, the pull request contains the necessary addition a cookie into all requests in which Ruuter queries itself across all of the DSLs in use throughout the Training Module. This ensures that the cookie is not lost and gets properly carried between cycles of requests, all of which trigger the .guard files that require it.

- Added and cleaned up the SQL and front-end query for last modified intent data.
- Added request headers to all instances in which Ruuter queries itself to keep carrying the cookie.

**Comment by Kristjan259** [comment]
When I have modified the intent (like added an example) it does not whos the date.



![Image](https://github.com/buerokratt/Training-Module/assets/5470201/85b7d2aa-5bcc-4d44-a872-83e482cbc7c9)

**Comment by KasparMinn** [comment]
After some troubleshooting with the help of Varmo and Alar, we managed to fix the database connection in the DEV environment and the date should now work.

**Important:** Since the correct data is not written for **old** (created before fix) intents, please verify the task with newly created intents and the fetched _last-modified_ should be correct.

**Comment by ffrose** [comment]
BUG - unable to see the correct data. After creating new intents, it does not show correct last-modified information. 



![Image](https://github.com/buerokratt/Training-Module/assets/119657383/e408414f-3b17-42f2-8d5a-c8fb92d2e668)
